core/src/apps/ontology/sc/native_builder.py

```diff
@@ -0,0 +1,120 @@
+from ubinascii import unhexlify
+
+from trezor import wire
+
+from .. import writer
+from . import builder, opcode
+
+
+class ParamStruct:
+    """
+    Special struct for smart contract argument passing
+    """
+
+    def __init__(self, arr: list):
+        self.arr = arr
+
+
+def build_native_call(func_name: str, params: list, contract: bytes) -> bytes:
+    """
+    Builds native contract call
+    """
+    ret = bytearray()
+
+    _write_native_code_script(ret, params)
+    builder.write_push_bytes(ret, func_name.encode())
+    builder.write_push_bytes(ret, contract)
+    builder.write_push_int(ret, 0)
+    writer.write_byte(ret, opcode.SYSCALL)
+    builder.write_push_bytes(ret, b"Ontology.Native.Invoke")
+
+    return ret
+
+
+def _write_native_code_script(ret: bytearray, arr: list) -> None:
+    """
+    Writes native code script from supplied data
+    """
+    for val in reversed(arr):
+        if isinstance(val, str):
```

Comment:
is this branch used anywhere?
using hex-encoded string internally is highly suspicious

Reply:
it doesn't seem to be, at least not in the device tests

Reply:
Should I remove the branch?